lib/fizzy/instructions.hpp

```diff
@@ -16,6 +16,18 @@ struct InstructionMetrics
     /// The stack height change caused by the instruction execution,
     /// i.e. stack height _after_ execution - stack height _before_ execution.
     int8_t stack_height_change;
+
+    /// The memory width in bytes the instruction operates on or 0 in case not using the memory.
+    uint8_t memory_width;
```

Comment:
I can see now that it would be simpler to keep this as `max_align` of `{1,2,3,4}`, as in your tests.

Reply:
Maybe, but `memory_width` is also fitting and describes what it does.

Reply:
I mean `{0,1,2,3}` so you don't need to pre-process it during validation:
```
            if (align > metrics.max_align)
                throw validation_error{"alignment cannot exceed operand size"};
```

Reply:
I see. Even if we change that I still wouldn't use `instruction_metrics` in the tests, otherwise the tests aren't that prone to mistakes :)

Reply:
Agreed.

Reply:
```
    /// The largest acceptable alignment value satisfying `2 ** max_align < memory_width` where memory_width
    /// is the number of bytes the instruction operates on. max_align is 0 if it does not operate on memory.
    uint8_t max_align;
```

This is actually a bit weird because `max_align = 0` is a valid value and with the current way `memory_width = 0` would be invalid which is the default on non-memory instructions. It is only a nitpick. The difference in the parser is a single shift less.

Reply:
Not an issue to me.